recipe/patch_yaml/conda-smithy.yaml

```
@@ -42,3 +42,11 @@ then:
   - tighten_depends:
       name: conda-build
       upper_bound: 24.5.0
+---
+if:
+  name: conda-smithy
+  version_le: 3.37.1
+then:
+  - tighten_depends:
+      name: conda-build
+      upper_bound: 24.7.1
```

as a minor nit, we could make it 24.7.0 (even though that this is probably never going to be released in conda-forge, it would break in the same way)
```suggestion
      upper_bound: 24.7.0
```

---

Good idea. Fixed.